bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -70,12 +73,23 @@ def apply_prep(target_spec)
 
           unless need_install.empty?
             need_install_targets = need_install.map(&:target)
-            run_task(executor, need_install_targets, 'puppet_agent::install')
-            # Service task works best when targets have puppet-agent feature
+            # lazy-load expensive gem code
+            require 'concurrent'
+            pool = Concurrent::ThreadPoolExecutor.new
+
+            futures = need_install_targets.map do |t|
+              opts = t.plugin_hooks&.fetch('puppet_library')
+              inventory.plugins.get_hook(opts['plugin'], :load_task).call(opts)
```

Comment:
if this fails any previously started tasks are orphaned. either we need to do this(and get the `puppet_library`) in a separate loop from creating the future or we need to wrap it in a `begin`/`rescue` and make a failed result for each of these.

Reply:
We also probably need to catch the error here and either ignore it, assuming the plugin doesn't have to prep anything in serial.  Otherwise this generates a confusing error about the `load_task` hook which the user has never heard about.